src/rpdk/core/jsonutils/utils.py

```diff
@@ -138,11 +140,17 @@ def schema_merge(target, src, path):  # noqa: C901
             try:
                 target[key] = schema_merge(target_schema, src_schema, next_path)
             except TypeError:
-                if key == "type":
-                    if isinstance(target_schema, list):
-                        target_schema.append(src_schema)
-                        continue
-                    target[key] = [target_schema, src_schema]
+                if key in (TYPE, REF):  # combining multiple $ref and types
```

Comment:
what happens if key is ref here? handled the same as multiple types?

Reply:
```
>>> a, b = {'Foo': {'$ref': 'a'}}, {'Foo': {'$ref': 'b'}}
>>> schema_merge(a, b, ('foo',))
{'Foo': { 'type': [(..., ...), (..., ...)]}}
```